### DISCUSSION TITLE
Upgrading from NativeScript Core from v5 - Cannot Navigate to page

### DISCUSSION THREAD

**Discussion by SaadmanChowdhury** [NativeScript/NativeScript#9655] [Q&A]
Good evening,

I'm trying to upgrade a version 5 NativeScript project to version 8.
I began by first upgrading the project to 6, 
and then testing if all of the functionalities work before moving on to upgrading it further.
After TNS migrate, and resolving a few minor issues, I hit a snag.

After the login screen, when trying to navigate to another screen, the **topmost()** returns **undefined**.

I've used the debugger and realized that the **frame_stack_1** in **frame-common.js** is empty.
I think the reason is that the frames are not being pushed into frame_stack_1 when the app loads.

I checked out this guide 
https://nativescripting.com/posts/missing-frame-and-page
`platformNativeScriptDynamic({ createFrameOnBootstrap: true }).bootstrapModule(
  AppModule
);`
but it is for angular.

Is there was a similar method of making it work for NativeScript Core?

- I'm a new comer to NativeScript, and I would appreciate and help at all. ^_^

**Comment by samiul-gtmi** [comment]
I am also facing the same problem. 
I'm trying to upgrade my client's project from v5 to v6, the same problem kept me busy whole week without a single workable solution online.
I'd really like a response to this problem.

**Comment by rigor789** [comment]
@SaadmanChowdhury @samiul-gtmi please provide more context in order for anyone to be able to help - code snippets, stack-traces etc.

**Comment by OPADA-Eng** [comment]
The breaking changes between 5 and 8 are the key to upgrade successfully.
I suggest you clone a fresh project with the new dependencies and try to add your plugins one by one.
finally, move your code and try to modify the breaking changes.

**Comment by dlcole** [comment]
You likely need to use Frame.topmost() 

```
const Frame = require("tns-core-modules/ui/frame").Frame;
Frame.topmost().navigate(navigationOptions);
```

When you get to later versions of {N} the first line becomes 

`import {Frame} from "@nativescript/core/";`
